File: tests/testdata/codegen/config-variables-pp/nodejs/config-variables.ts
import * as pulumi from "@pulumi/pulumi";

const config = new pulumi.Config();
const requiredString = config.require("requiredString");
const requiredInt = config.requireNumber("requiredInt");
const requiredFloat = config.requireNumber("requiredFloat");
const requiredBool = config.requireBoolean("requiredBool");
const requiredAny = config.requireObject<any>("requiredAny");
const optionalString = config.get("optionalString") || "defaultStringValue";
const optionalInt = config.getNumber("optionalInt") || 42;
const optionalFloat = config.getNumber("optionalFloat") || 3.14;
const optionalBool = config.getBoolean("optionalBool") || true;
const optionalAny = config.getObject<any>("optionalAny") || {
    key: "value",
};

File: tests/testdata/codegen/config-variables-pp/go/config-variables.go
package main

import (
	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
	"github.com/pulumi/pulumi/sdk/v3/go/pulumi/config"
)

func main() {
	pulumi.Run(func(ctx *pulumi.Context) error {
		cfg := config.New(ctx, "")
		requiredString := cfg.Require("requiredString")
		requiredInt := cfg.RequireInt("requiredInt")
		requiredFloat := cfg.RequireFloat64("requiredFloat")
		requiredBool := cfg.RequireBool("requiredBool")
		requiredAny := cfg.RequireObject("requiredAny")
		optionalString := "defaultStringValue"
		if param := cfg.Get("optionalString"); param != "" {
			optionalString = param
		}
		optionalInt := 42
		if param := cfg.GetInt("optionalInt"); param != 0 {
			optionalInt = param
		}
		optionalFloat := float64(3.14)
		if param := cfg.GetFloat64("optionalFloat"); param != 0 {
			optionalFloat = param
		}
		optionalBool := true
		if param := cfg.GetBool("optionalBool"); param {
			optionalBool = param
		}
		optionalAny := map[string]interface{}{
			"key": "value",
		}
		if param := cfg.GetObject("optionalAny"); param != nil {
			optionalAny = param
		}
		return nil
	})
}
